File: CVerify/CVerify.AI/CVerify/CVerify.AI/Skills/SkillNormalizer.cs
namespace CVerify.AI.Skills;

public class SkillNormalizer
{
    private readonly ISkillOntology _ontology;

    public SkillNormalizer(ISkillOntology ontology)
    {
        _ontology = ontology;
    }

    public async Task<string> NormalizeAsync(string skillName)
    {
        // Implementation will go here - map skill to ontology
        return skillName;
    }

    public async Task<string[]> NormalizeMultipleAsync(string[] skills)
    {
        // Implementation will go here
        return skills;
    }
}

File: CVerify/CVerify.AI/app/skills/skill_normalizer.py
from app.skills.skill_ontology import ISkillOntology


class SkillNormalizer:
    def __init__(self, ontology: ISkillOntology):
        self._ontology = ontology

    async def normalize_async(self, skill_name: str) -> str:
        definition = await self._ontology.get_skill_async(skill_name)
        return definition.name if definition.name else skill_name

    async def normalize_multiple_async(self, skills: list[str]) -> list[str]:
        return [await self.normalize_async(s) for s in skills]
